pkg/myofficefiles/manager.go

```diff
@@ -0,0 +1,99 @@
+package myofficefiles
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strconv"
+
+	gatewayv1beta1 "github.com/cs3org/go-cs3apis/cs3/gateway/v1beta1"
+	userpb "github.com/cs3org/go-cs3apis/cs3/identity/user/v1beta1"
+	rpc "github.com/cs3org/go-cs3apis/cs3/rpc/v1beta1"
+	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+	typesv1beta1 "github.com/cs3org/go-cs3apis/cs3/types/v1beta1"
+	"github.com/cs3org/reva/pkg/appctx"
+	"github.com/cs3org/reva/pkg/errtypes"
+	"github.com/cs3org/reva/pkg/rgrpc/todo/pool"
+	"github.com/cs3org/reva/pkg/storage/utils/templates"
+)
+
+// Manager defines an interface for a MyOfficeFiles manager.
+type Manager interface {
+	// ListMyOfficeFiles returns all recent Office files of a user.
+	ListMyOfficeFiles(ctx context.Context, user *userpb.User) ([]*provider.ResourceInfo, error)
+}
+
+// This feature is only enabled for users that are in the
+const (
+	targetGroup      = "cernbox-office-view"
+	officeFilesRegex = "(.*?)(.xls|.doc|.ppt|.XLS|.DOC|.PPT)[x|X]?"
```

Comment:
The feature ticket requested that the filtering is done by type of office file. So our search should not return all office files combined in a single list, but instead by powerpoint OR excel OR word files.

Reply:
Also, the regex is not matching the extension with the end of the filename.